cmd/snap-update-ns/utils.go

```diff
@@ -62,10 +66,92 @@ type ReadOnlyFsError struct {
 	Path string
 }
 
+// Error returns a formatted error message.
 func (e *ReadOnlyFsError) Error() string {
 	return fmt.Sprintf("cannot operate on read-only filesystem at %s", e.Path)
 }
 
+// MimicPath returns the path of the directory where a mimic ought to be constructed.
+func (e *ReadOnlyFsError) MimicPath() string {
+	return e.Path
+}
+
+// TrespassingError is an error when filesystem operation would affect the host.
+type TrespassingError struct {
+	Path string
+}
+
+// MimicPath returns the path of the directory where a mimic ought to be constructed.
+func (e *TrespassingError) Error() string {
+	return fmt.Sprintf("cannot trespass on host filesystem at %s", e.Path)
+}
+
+// MimicPath returns the path of the directory where a mimic ought to be constructed.
+func (e *TrespassingError) MimicPath() string {
+	return e.Path
+}
+
+// TODO: Handle /home/*/snap/* when we do per-user mount namespaces and allow
+// defining layout items that refer to SNAP_USER_DATA and SNAP_USER_COMMON.
+//
+// Note that /snap is "not allowed" because it is handled by existing EROFS code.
+// and thus doesn't need any extra logic.
+var allowedPaths = []string{"/var/snap/", "/tmp/", "/snap/$SNAP_NAME/$SNAP_REVISION/"}
+
+// MimicRequiredError describes errors that require construction of a writable mimic.
+type MimicRequiredError interface {
+	Error() string
+	MimicPath() string
+}
+
+// checkTrespassing returns an error if path cannot be written to without a mimic.
```

Comment:
The first line describing the function needs to say what it does, not what the potential error is.

Reply:
Fixed now